### PR TITLE
New Rule - Depth Level.

### PR DESCRIPTION
**Pull request by jcalvor**
New Rule - Depth Level. Sets the depth of nested styles in every parent
style.

### CODE REVIEW

File: lib/linters/depth_level.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      if (error) {
+          return [{
+              message: util.format(this.message, depth)
+          }];
+      }
+    }
+};
+
+/*Checks how many nested classes have every node*/
+var countLevels = function (childNode,i, total) {
```
Comment (shellscape) [inline lib/linters/depth_level.js]: argument spacing

File: lib/linters/depth_level.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  total += 1;
+  var block = childNode.first('block') || {};
+  block.eachFor('ruleset',function(subchildNode,i) {
```
Comment (shellscape) [inline lib/linters/depth_level.js]: argument spacing

File: lib/linters/depth_level.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+*/
+module.exports = {
+    name: 'depthLevel',
+    nodeTypes: ['block'],
+    message: "There shouldn't be more than '%s' levels deep from the style's parent, check the children's depth.",
+
+    lint: function deepLevelLinter (config, node) {
+      let depth = config.depth || 2;
+      let error = false;
+      node.eachFor('ruleset',function(childNode,i) {
```
Comment (shellscape) [inline lib/linters/depth_level.js]: argument spacing

**Comment by shellscape** [comment]
Looks good to me, but missing some things before the PR can be approved.

Need to create tests for the linter.
Need to create the readme for the linter and update the linters TOC to point to it as well.

You've got a bunch of linting errors in the code though. see https://travis-ci.org/lesshint/lesshint/jobs/110491786. Please run linting and tests locally as well.

**Comment by jwilsson** [comment]
@shellscape beat me to it :)

It looks good, just the things he mentioned.

**Comment by jcalvor** [comment]
O,k thanks for the feedback. I made some changes to fix the lint errors. I also created a test and updated the readme doc. Let me know if every is ok or if I need to make a change.

**Comment by kokarn** [comment]
Only thing I can think of is that maybe this should be disabled as a default instead of 3? 
Thoughts?

**Comment by CITguy** [comment]
Agree with @kokarn on disabled by default. However, I'd name it `validateIndentation` as suggested in #50.

**Comment by CITguy** [comment]
Also, how is this different from #107?

**Comment by kokarn** [comment]
Wow, yeah that's true. Is #107 ready to merge @jwilsson or should we go ahead with this?

**Comment by jwilsson** [comment]
@CITguy @kokarn This PR checks the number of nested selectors while #107 (which isn't ready yet, it's a bit trickier then I originally thought) only checks your indentation.

It seems like the majority thinks this should be disabled by default, and I can see the reasoning behind that too, I think we should merge it and just set it to disabled.

**Comment by kokarn** [comment]
If that's what you want and don't feel #107 is done, let's do that. Then maybe we can just improve on this at a later date.

**Comment by jwilsson** [comment]
Huge thanks for this @jcalvor! :tada:

**Comment by jcalvor** [comment]
Thank you guys! This is a great project and i hope this can help others.

**Pull request merged**